core/src/main/java/com/linecorp/armeria/internal/common/DefaultCancellationScheduler.java

```diff
@@ -76,299 +58,233 @@ final class DefaultCancellationScheduler implements CancellationScheduler {
     private long startTimeNanos;
     @Nullable
     private EventExecutor eventLoop;
-    @Nullable
-    private CancellationTask task;
-    @Nullable
-    private volatile Runnable pendingTask;
+    private volatile CancellationTask task = noopCancellationTask;
     @Nullable
     private ScheduledFuture<?> scheduledFuture;
+    private long setFromNowStartNanos;
+    private TimeoutMode timeoutMode = TimeoutMode.SET_FROM_START;
+    @Nullable
+    private volatile Throwable cause;
+    private final Ticker ticker;
+    private final ReentrantShortLock lock = new ReentrantShortLock();
+
+    private final boolean server;
     @Nullable
     private volatile CancellationFuture whenCancelling;
     @Nullable
     private volatile CancellationFuture whenCancelled;
```

Comment:
Because `whenCancelled()`, which creates this future, is called almost immediatly after this instance is created, how about creating this future directory at construction? 
`private final CancellationFuture whenCancelled = new CancellationFuture();`